x/evm/keeper/msg_server.go

```diff
@@ -176,21 +174,27 @@ func (server msgServer) EVMTransaction(goCtx context.Context, msg *types.MsgEVMT
 			},
 		)
 
-	} else {
-		// if applyErr is nil then res must be non-nil
-		if res.Err != nil {
-			serverRes.VmError = res.Err.Error()
+		return
```

Comment:
in this case, we still would want a hash in the tx response to be consistent with the previous implementation, right? in this version, it would now return `nil` for serverRes instead of a struct with the hash field set